Sources/StreamChat/APIClient/Endpoints/Payloads/UserPayloads.swift

```diff
@@ -40,6 +40,27 @@ class UserPayload<ExtraData: UserExtraData>: Decodable {
     let teams: [String]
     let extraData: ExtraData
     
+    var hasher: ChatHasher {
+        UserHasher(
+            id: id,
+            name: name,
+            imageURL: imageURL,
+            role: role.rawValue,
+            createdAt: createdAt,
+            updatedAt: updatedAt,
+            lastActiveAt: lastActiveAt,
+            isOnline: isOnline,
+            isBanned: isBanned,
+            extraData: (try? JSONEncoder.default.encode(extraData)) ?? .init()
+        )
+    }
+    
+    // This needs to be explicit so we can override it in
+    // `UserDTO_Tests.test_DTO_skipsUnnecessarySave`
+    var chatHash: Int {
```

Comment:
nit: why `ChatHash`? :) Isn't' everything "Chat" when we're building a chat SDK? 🤔 `changesHash` `versionHash`?

Reply:
My mind slipped to our own custom hash (instead of built-in, well known hash), another approach of mine was `updateHash` but I like `changesHash` too

Reply:
fixed in d1a4ed6f97a6a61cec7487f6fe07b07120e78617